packages/nuxt/src/app/composables/hydrate.ts

```diff
@@ -1,6 +1,36 @@
 import { useNuxtApp } from '../nuxt'
 import type { NuxtPayload } from '../nuxt'
 
+/**
+ * An SSR-friendly utility to call a method once
+ * @param key a unique key ensuring the function can be properly de-duplicated across requests
+ * @param fn a function to call
+ * @see https://nuxt.com/docs/api/utils/once
+ */
+export function once (key?: string, fn?: (() => any | Promise<any>)): Promise<void>
+export function once (fn?: (() => any | Promise<any>)): Promise<void>
+export async function once (...args: any): Promise<void> {
+  const autoKey = typeof args[args.length - 1] === 'string' ? args.pop() : undefined
+  if (typeof args[0] !== 'string') { args.unshift(autoKey) }
+  const [_key, fn] = args as [string, (() => any | Promise<any>)]
+  if (!_key || typeof _key !== 'string') {
+    throw new TypeError('[nuxt] [once] key must be a string: ' + _key)
+  }
+  if (fn !== undefined && typeof fn !== 'function') {
+    throw new Error('[nuxt] [once] fn must be a function: ' + fn)
```

Comment:
Is there a reason this one is an `Error` while the one above is a `TypeError`?

Reply:
It is similar to https://github.com/nuxt/nuxt/blob/a82a724baafb18dca0b07e3e6a7d31d54cec95e6/packages/nuxt/src/app/composables/state.ts#L21

For me I am ok to update it to TypeError, let's see what @danielroe thinks about it